tests/array15.dora

```diff
@@ -0,0 +1,53 @@
+fun main() {
+
+    let x = arrayFill::<Bool>(3, true);
+    assert(arrayContains::<Bool>(x, true));
+    assert(arrayHas::<Bool>(x, true));
+
+    let x = arrayFill::<Int>(3, 3);
+    assert(arrayContains::<Int>(x, 3));
+    assert(arrayHas::<Int>(x, 3));
+
+    let x = arrayFill::<Long>(3, 3L);
+    assert(arrayContains::<Long>(x, 3L));
+    assert(arrayHas::<Long>(x, 3L));
+
+    // let x = arrayFill::<Double>(3, 0.0/0.0);
```

Comment:
I can look into this if you want, could you please open a bug with this?

Reply:
Yes, this would be great! I'm still very bad at debugging this register-related stuff.

Reply:
https://github.com/dinfuehr/dora/issues/59

Reply:
Yes, this is a bit messy in the compiler. We might have bytecode in the baseline compiler in the future, hopefully this will be a bit simpler than.